src/kubernetes-ingress-controller/guides/using-kongingress-resource.md

```diff
@@ -120,132 +139,210 @@ $ echo "apiVersion: configuration.konghq.com/v1
+	pod name:	echo-588c888c78-6jrmn
+	pod name:	echo-588c888c78-6jrmn
+	pod name:	echo-588c888c78-6jrmn
+	pod name:	echo-588c888c78-6jrmn
+```
 
-Hostname: echo-d778ffcd8-vrrtw
+If we then add the header, Kong will hash its value and distribute it to the
+same replica when we use the same value:
 
-Pod Information:
-  node name:	gke-harry-k8s-dev-default-pool-bb23a167-8pgh
-  pod name:	echo-d778ffcd8-vrrtw
-  pod namespace:	default
-  pod IP:	10.60.0.9
+```bash
+$ for n in {1..3}; do
+  curl -s $PROXY_IP/foo -H "x-lb: foo" | grep "pod name:";
+  curl -s $PROXY_IP/foo -H "x-lb: bar" | grep "pod name:";
+  curl -s $PROXY_IP/foo -H "x-lb: baz" | grep "pod name:";
+done
+	pod name:	echo-588c888c78-nk4jc
+	pod name:	echo-588c888c78-nk4jc
+	pod name:	echo-588c888c78-6jrmn
+	pod name:	echo-588c888c78-nk4jc
+	pod name:	echo-588c888c78-nk4jc
+	pod name:	echo-588c888c78-6jrmn
+	pod name:	echo-588c888c78-nk4jc
+	pod name:	echo-588c888c78-nk4jc
+	pod name:	echo-588c888c78-6jrmn
+```
 
-Server values:
-  server_version=nginx: 1.12.2 - lua: 10010
+Increasing the replicas will redistribute some subsequent requests onto the new
+replica:
 
-Request Information:
-  client_address=10.60.1.10
-  method=GET
-  real path=/baz
-  query=
-  request_version=1.1
-  request_scheme=http
-  request_uri=http://35.233.170.67:8080/baz
+```bash
+$ kubectl patch deploy echo --patch '{"spec": {"replicas": 3}}'
+deployment.apps/echo patched
 ```
 
-As you can see, the real path value is `/baz`.
+```bash
+$ for n in {1..3}; do
+  curl -s $PROXY_IP/foo -H "x-lb: foo" | grep "pod name:";
+  curl -s $PROXY_IP/foo -H "x-lb: bar" | grep "pod name:";
+  curl -s $PROXY_IP/foo -H "x-lb: baz" | grep "pod name:";
+done
+	pod name:	echo-588c888c78-nk4jc
+	pod name:	echo-588c888c78-d477r
+	pod name:	echo-588c888c78-6jrmn
+	pod name:	echo-588c888c78-nk4jc
+	pod name:	echo-588c888c78-d477r
+	pod name:	echo-588c888c78-6jrmn
+	pod name:	echo-588c888c78-nk4jc
+	pod name:	echo-588c888c78-d477r
+	pod name:	echo-588c888c78-6jrmn
+```
 
-## Use KongIngress with Service resource
+Kong's load balancer does not directly distribute requests to each of the
+Service's Endpoints. It first distributes them evenly across a number of
+equal-size buckets. These buckets are then distributed across the available
+Endpoints according to their weight. For Ingresses, however, there is a single
+Service, and the controller assigns each Endpoint (represented by a Kong
+upstream target) equal weight, and requests are indeed evenly hashed across all
+Endpoints.
+
+Gateway API HTTPRoutes support multiple backend Services with associated
+weights per Service. The controller still assigns each Endpoint for a given
+Service the same weight, but assigns them proportionally across the
+multi-Service backend: the Endpoints of a Service will have a Kong target
+weight such they collectively receive a percentage of requests equal to their
+Service's weight in the HTTPRoute. For example, if you have two Services, one
```

Comment:
Everyone loves jargon-filled math equations written in prose. Rewrote it, not sure if it's any better. Expanded the example to show changes, since IMO the example probably explains it better through demonstration than the explanation does.